### ISSUE TITLE
block attribute copies after the instance as been created by the rule

### ISSUE THREAD

**Issue by levilucio** [mbeddr/language_verification#27]
- Write a rule A where an instance is created
- Write a rule B in a subsequent layer where an attribute of the rule is set
- This rule will fail in DSLTrans, so attribute creation after the instance has been created should appear marked as an error

**Comment by BentleyJOakes** [comment]
Please refresh my memory. Can attribute creation occur on an element connected to a backward link? It already exists...

**Comment by clagms** [comment]
Hi Bentley!
I think it cannot.
If attributes are used, it is to used them to match the element.

Cheers!

**Comment by BentleyJOakes** [comment]
Isn't this connected to the two semantics of the backward links? If it's on the 'matching backward link', then it's a check for the attribute. If it's connected to the 'creation backward link', then it's attribute creation.

**Comment by clagms** [comment]
Yes it is. I though your were referring to the DSLTrans backward links, and
not to the backward links in the contract language.

**Comment by levilucio** [comment]
The DSLTrans engine throws an exception when the attribute is set after the the instance is created. So with the current implementation all attribute setting has to be done at the time of instance creation.
